optaplanner-core/src/main/java/org/optaplanner/core/config/score/director/ScoreDirectorFactoryConfig.java

```diff
@@ -169,12 +171,22 @@ public Boolean getDroolsAlphaNetworkCompilationEnabled() {
         return droolsAlphaNetworkCompilationEnabled;
     }
 
+    @Deprecated
     public KieRuntimeBuilderWrapper getGizmoKieRuntimeBuilderWrapper() {
-        return gizmoKieRuntimeBuilderWrapper;
+        return (KieRuntimeBuilderWrapper) getKieBaseSupplier();
     }
 
+    @Deprecated
     public void setGizmoKieRuntimeBuilderWrapper(KieRuntimeBuilderWrapper kieRuntimeBuilderWrapper) {
-        this.gizmoKieRuntimeBuilderWrapper = kieRuntimeBuilderWrapper;
+        setKieBaseSupplier(kieRuntimeBuilderWrapper);
+    }
+
+    public Supplier<KieBase> getKieBaseSupplier() {
```

Comment:
Maybe we rename to `...GizmoKieBaseSupplier`, so that the point becomes more obvious?